core/src/main/java/de/jplag/merging/MatchMerging.java

```diff
@@ -110,6 +113,46 @@ private List<Match> mergeNeighbors(List<Match> globalMatches, Submission leftSub
         return globalMatches;
     }
 
+    /**
+     * This function checks if a merge would go over file boundaries.
+     * @param leftSubmission is the left submission
+     * @param rightSubmission is the right submission
+     * @param upperNeighbor is the upper neighboring match
+     * @param tokensBetweenLeft amount of token that separate the neighboring matches in the left submission and need to be
+     * removed
+     * @param tokensBetweenRight amount token that separate the neighboring matches in the send submission and need to be
+     * removed
+     * @return true if the merge goes over file boundaries.
+     */
+    private boolean mergeOverlapsFiles(Submission leftSubmission, Submission rightSubmission, Match upperNeighbor, int tokensBetweenLeft,
+            int tokensBetweenRight) {
+        if (leftSubmission.getFiles().size() == 1 && rightSubmission.getFiles().size() == 1) {
+            return false;
+        }
+        int startLeft = upperNeighbor.startOfFirst();
+        int startRight = upperNeighbor.startOfSecond();
+        int lengthUpper = upperNeighbor.length();
+
+        List<Token> tokenLeft = new ArrayList<>(leftSubmission.getTokenList());
+        List<Token> tokenRight = new ArrayList<>(rightSubmission.getTokenList());
+        tokenLeft = tokenLeft.subList(startLeft + lengthUpper, startLeft + lengthUpper + tokensBetweenLeft);
+        tokenRight = tokenRight.subList(startRight + lengthUpper, startRight + lengthUpper + tokensBetweenRight);
+
+        for (Token token : tokenLeft) {
+            if (token.getType().equals(SharedTokenType.FILE_END)) {
+                return true;
+            }
+        }
+
+        for (Token token : tokenRight) {
+            if (token.getType().equals(SharedTokenType.FILE_END)) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
```

Comment:
Could be also:
```java
return containsFileEndToken(tokenLeft) || containsFileEndToken(tokenRight) 
```

with
```java
private boolean containsFileEndToken(List<Token> token) {
    return token.stream().map(Token::getType).anyMatch(it->it.equals(SharedTokenType.FILE_END);
}
```

(reduces duplication, makes it more compact)

Reply:
Does containsFileEndToken need JDoc?

Reply:
No, I think it is self-explaining enough.